examples/github/main.tf

```diff
@@ -66,26 +73,33 @@ data "kubectl_file_documents" "install" {
   content = data.flux_install.main.content
 }
 
-resource "kubectl_manifest" "install" {
-  for_each   = { for v in data.kubectl_file_documents.install.documents : sha1(v) => v }
-  depends_on = [kubernetes_namespace.flux_system]
-
-  yaml_body = each.value
-}
-
 data "kubectl_file_documents" "sync" {
   content = data.flux_sync.main.content
 }
 
-resource "kubectl_manifest" "sync" {
-  for_each   = { for v in data.kubectl_file_documents.sync.documents : sha1(v) => v }
-  depends_on = [kubectl_manifest.install, kubernetes_namespace.flux_system]
+locals {
+  install = [ for v in data.kubectl_file_documents.install.documents : {
+      data: yamldecode(v)
+      content: v
+    }
+  ]
+  sync = [ for v in data.kubectl_file_documents.sync.documents : {
+      data: yamldecode(v)
+      content: v
+    }
+  ]
+}
 
+resource "kubectl_manifest" "install" {
+  for_each   = { for v in local.install : lower(join("/", compact([v.data.apiVersion, v.data.kind, lookup(v.data.metadata, "namespace", ""), v.data.metadata.name]))) => v.content }
+  depends_on = [kubernetes_namespace.flux_system]
   yaml_body = each.value
```

Comment:
Will changes to yaml_body trigger a replacement? Is there any risk of the compact statement staying the same but body changing - but not being applied?

Reply:
Nope so I have tried upgrading from multiple different versions. The only thing the key does is make sure that the kubectl provider updates resources in place instead of replacing them, when the yaml content changes.